upcloud/account.go

```diff
@@ -38,3 +44,138 @@ func (s *Account) UnmarshalJSON(b []byte) error {
 
 	return nil
 }
+
+// AccountCampaigns represents campaigns associated to account
+type AccountCampaigns struct {
+	Campaign []string `json:"campaign"`
+}
+
+// AccountRoles represents roles associated to account
+// Roles for the account; billing, aux_billing, or technical.
+type AccountRoles struct {
+	Role []string `json:"role"`
+}
+
+// AccountTag represents tag and storage access permisission
+type AccountTag struct {
+	Name    string  `json:"name"`
+	Storage Boolean `json:"storage"`
+}
+
+// AccountTagAccess represents tags associated to account
+type AccountTagAccess struct {
+	Tag []AccountTag `json:"tag"`
+}
+
+// AccountServer represents server (UUID) and storage access permisission
+type AccountServer struct {
+	UUID    string  `json:"uuid"`
+	Storage Boolean `json:"storage"`
+}
+
+// AccountServerAccess represents servers the account is allowed to manage
+type AccountServerAccess struct {
+	Server []AccountServer `json:"server"`
+}
+
+// AccountStorageAccess represents UUIDs of storages the account is allowed to manage, or * for all
+type AccountStorageAccess struct {
+	Storage []string `json:"storage"`
+}
+
+// AccountNetworkAccess represents UUIDs of networks the account is allowed to manage, * means all
+type AccountNetworkAccess struct {
+	Network []string `json:"network"`
+}
+
+// AccountIPFilters represents IP address restrictions on API access; if set, allowed only from the specified ranges.
+// Ranges can be specified in CIDR format, ranges separated by a dash, or as single IP addresses
+type AccountIPFilters struct {
+	IPFilter []string `json:"ip_filter"`
+}
+
+// AccountDetails represents detailed information about an account
+type AccountDetails struct {
+	MainAccount string `json:"main_account"`
+	Type        string `json:"type"`
```

Comment:
Would it make sense to make a separate enum for type? I think it's useful to narrow the types as much as possible so that the future developers don't need to wonder what values other then "sub" are allowed here

Reply:
Good point I'll change that